monai/networks/nets/quicknat.py

```diff
@@ -0,0 +1,222 @@
+        max_pool = get_pool_layer(
+            name=("max", {"kernel_size": pool, "stride": stride_pool, "return_indices": True, "ceil_mode": True}),
+            spatial_dims=2,
+        )
+        # for the unpooling layer there is currently no Monai implementation available, return to torch implementation
+        un_pool = nn.MaxUnpool2d(kernel_size=pool, stride=stride_pool)
+
+        # sequence of convolutional strides (like in UNet) not needed as they are always stride_conv. This defaults to 1.
+        def _create_model(layer: int) -> nn.Module:
+            """
+            Builds the QuickNAT structure from the bottom up by recursing down to the bottelneck layer, then creating sequential
+            blocks containing the decoder, a skip connection around the previous block, and the encoder.
+            At the last layer a classifier block is added to the Sequential.
+
+            Args:
+                layer = inversproportional to the layers left to create
+            """
+            subblock: nn.Module
+            if layer < 4:
+                subblock = _create_model(layer + 1)
+
+            else:
+                subblock = Bottleneck(num_filters, se_layer, dropout_layer, max_pool, un_pool, kernel_size, num_filters)
+
+            if layer == 1:
+                down = ConvConcatDenseBlock(num_channels, se_layer, dropout_layer, kernel_size, num_filters)
+                up = ConvConcatDenseBlock(num_filters * 2, se_layer, dropout_layer, kernel_size, num_filters)
+                classifier = ClassifierBlock(2, num_filters, num_classes, stride_conv, kernel_c)
+                return SequentialWithIdx(down, SkipConnectionWithIdx(subblock), up, classifier)
+            else:
+                up = Decoder(num_filters * 2, un_pool, se_layer, dropout_layer, kernel_size, num_filters)
+                down = Encoder(num_filters, max_pool, se_layer, dropout_layer, kernel_size, num_filters)
+                return SequentialWithIdx(down, SkipConnectionWithIdx(subblock), up)
+
+        self.model = _create_model(1)
+
+    def get_selayer(self, n_filters, se_block_type="None"):
+        """
+        Returns the SEBlock defined in the initialization of the QuickNAT model.
+
+        Args:
+            n_filters: encoding half of the layer
+            se_block_type: defaults to None. Valid options are None, CSE, SSE, CSSE
+        Returns: Appropriate SEBlock. SSE and CSSE not implemented in Monai yet.
+        """
+        if se_block_type == "CSE":
+            return se.ChannelSELayer(2, n_filters)
+        # not implemented in squeeze_and_excitation in monai use other squeeze_and_excitation import:
+        elif se_block_type == "SSE" or se_block_type == "CSSE":
+            # Throw error if squeeze_and_excitation is not installed
+            if not flag:
+                raise ImportError("Please install squeeze_and_excitation locally to use SpatialSELayer")
```

Comment:
@ericspod don't I break after raising the Error. I check in line 170 if se_block_type is "SSE" or "CSSE", if the import  now set in the beginning was not successful it should break. I don't see the necessity to repeat the check after line 176 maybe you could elaborate?